_posts/2017-11-03-2017-crystal-survey-2017-results.md

```diff
@@ -0,0 +1,230 @@
+"%63.1 Less than 1.000, %23 1000 to 10000 lines, %11.5 Not sure, %2.+ 10000 to 100000 line")
+(hover for more info)
+
+Most of the current Crystal projects are less than 1000 lines, However there are also 17 projects which are in between 10000 and 100000 lines.
+
+![%46.6 Yes, %32.9 Not sure, %11.9 No, %8.9 Not applicable](https://cdn.pbrd.co/images/GP39tRi.png
+  "%46.6 Yes, %32.9 Not sure, %11.9 No, %8.9 Not applicable")
+(hover for more info)
+
+Crystal is a beauty, we know that a lot of you are excited to use it for your daily work :)
+
+## What are some ways you or your company are using Crystal at work?
+
+220 people answered this free form question. Here are the most common answers to how companies are using Crystal
+
+- Microservices
+- CPU performance intensive apps
+- CLI tools
+- Background workers
+- Big Data processing
+- Ruby replacement
+
+Some very interesting ones
+
+- Genome analysis
+- HPC
+- Embedded GUI applications
+
+## How could we help make Crystal more accepted at your company?
+
+To make Crystal more accepted at companies, we've asked this question. We've got 370 responses
+
+- Stable release (1.0)
+- Parallelism
+- Windows support
+- More resources and documentation
+- Better tools (IDEs e.g)
+- More libraries
+- More streamlined release cycles
+
+## What editor(s) do you use when writing Crystal?
+
+Crystal developers do love using Vim, here's the major editors being used
+
+- Vim - %34
+- Atom - %32.2
+- Visual Studio Code - %28.8
+- Sublime Text - %28.4
+- Emacs - %8.1
+
+# Platforms
+
+![Linux %66, Mac OS %57.5, Windows %17.9, %2.8 BSD-variant](https://cdn.pbrd.co/images/GPbEV09.png
+  "Linux %66, Mac OS %57.5, Windows %17.9, %2.8 BSD-variant")
+(hover for more info)
+
+Linux is the main platform Crystal developers choose followed by Mac OS. There's also a notable amount (%17.9) of Crystal developers using Windows.
+
+![Linux %91.2, Mac OS %41.5, Windows %31.1, Android %13.1, iOS %10.9, %5.3 BSD-variant, %9.4 Javascript, %6 Embeeded](https://cdn.pbrd.co/images/GPbIN3w.png
+  "Linux %91.2, Mac OS %41.5, Windows %31.1, Android %13.1, iOS %10.9, %5.3 BSD-variant, %9.4 Javascript, %6 Embeeded")
+
+We see a similar pattern in target platforms. Linux is the most targeted platform. Interestingly we see Android, iOS and Javascript in a noticable ratio.
+
+# Towards 1.0
+
+![0.23.* %56.9, Not sure %22.4, Master %6.4](https://cdn.pbrd.co/images/GPbKSji.png
+  "0.23.* %56.9, Not sure %22.4, Master %6.4")
+(hover for more info)
+
+Crystal developers tend to follow the latest stable release.
+
+![No %60.7, Yes, but it happens rarely %29.8, Yes, and it happens oftern %9.4](https://cdn.pbrd.co/images/GPdzxCW.png
+  "No %60.7, Yes, but it happens rarely %29.8, Yes, and it happens oftern %9.4")
+(hover for more info)
+
+We try to keep breaking changes as minimum as possible. More than %60 of Crystal developers don't have any changes to their code when upgrading to a new version.

```

Comment:
This is pretty much a lie, we perform breaking changes all the time and with not much more reason than the change makes it better.

Reply:
Not everyone uses `HTTP::Handler`s and while maintaining Kemal for such a long time I haven't had many breaking changes after `0.20.0`. Just some minor stuff and compared to other pre-1.0  and heck even some 1.0+ languages it really feels minimum.

Reply:
But that we *try* to keep breaking changes to a minimum is I personally think a lie. We break things for minor API niceties like removing `post_form`. And i think that's a good thing but I don't want to lie about it.

An interesting statistic would be the pie chart about breaking changes limited to people who say they have 10000+ line codebases.

Reply:
I'd say that we're mindful of breaking changes but at this point we value cleaner APIs over backwards compatibility. And I think that's true, see for example the pre-0.24 discussions around `Time`.

Reply:
In the end no breaking change has been a pain to me: shards and my libraries seldom break, like once every 6 or 12 months, and are quickly fixed with a tiny patch. Even the 0.16 overhaul on types (all instance variables must be typed) was painless.

Reply:
Fir enough, but that's not true for anyone and neither was the statement about the *quality* of breaking changes, just the *quantity*.